wgpu/src/backend.rs

```diff
@@ -110,7 +110,13 @@ impl Backend {
         target_width: u32,
         target_height: u32,
     ) {
-        let bounds = (layer.bounds * scale_factor).snap();
+        let target_bounds = iced_graphics::Rectangle::with_size(
+            iced_graphics::Size::new(target_width as f32, target_height as f32),
+        );
+        let bounds = (layer.bounds * scale_factor)
+            .intersection(&target_bounds)
+            .unwrap_or(layer.bounds * scale_factor)
+            .snap();
```

Comment:
I am not sure we are tackling the issue directly with this fix.

Maybe we just need a couple of `min` calls in the `layer.bounds * scale_factor` result?

Reply:
Then maybe, this is better ?
```suggestion
        let mut bounds_float = layer.bounds * scale_factor;
        bounds_float.width = bounds_float.width.min(target_width as f32 - bounds_float.x);
        bounds_float.height = bounds_float.height.min(target_height as f32 - bounds_float.y);
        let bounds = bounds_float.snap();
```